File: d01/d01.py
def readlines(fn):
    with open(fn) as f:
        txt = f.read().splitlines()
    return txt

def part1(depths):
    return sum(map(lambda x,y: x<y, depths[:-1], depths[1:]))
    
def part2(depths):
    return part1([x+y+z for x,y,z in zip(depths[:-2], depths[1:-1], depths[2:])])

if __name__ == '__main__':
    fn = 'd01.txt'
    # fn = 'd01eg.txt'
    data = [int(x) for x in readlines(fn)]
    print(part1(data))
    print(part2(data))

File: d01/d01.jl
fn = "d01eg.txt"
fn = "d01.txt"
l = map(x -> parse(Int, x), readlines(fn))

part1(l) = sum(map( x -> x[1]<x[2], zip(l[1:end-1],l[2:end])))
@time part1(l)
@time part1(l)

part2(l) = part1(map(x -> sum(x), zip(l[1:end-2],l[2:end-1],l[3:end])))
@time part2(l)
@time part2(l)
